share/functions/fish_git_prompt.fish

```diff
@@ -211,6 +212,15 @@ function fish_git_prompt --description "Prompt function for Git"
 
     set -l space "$___fish_git_prompt_color$___fish_git_prompt_char_stateseparator$___fish_git_prompt_color_done"
 
+    # Suppress certain config options in untrusted repos.
+    set -l extra_config -c core.fsmonitor= -c diff.external=
```

Comment:
Today, overriding diff.external shouldn't make a difference.
- Our "git diff" already uses --no-ext-diff, which has the last word here
- "git diff-index" is a plumbing command that, by default, doesn't use external diff drivers,
  only if you add --ext-diff and --patch (without --patch it doesn't look at content changes)